src/interface.ts

```diff
@@ -15,33 +15,65 @@ enum ResolveError {
   ResolverTimeout = "ResolverTimeout"
 }
 
-interface Intent {
-  intent: IntentName;
-  context: Context;
+/**
+ * the result of calling desktopAgent.resolve() is `Promise<ResolveResults>`
+ * the ResolveResult contains the list of apps which support the intent
+ */
+export interface RaiseIntentResults {
+    data: any;
+}
+
+/**
+ * the result of calling desktopAgent.resolve() is `Promise<ResolveResults>`
+ * the ResolveResult contains the list of apps which support the intent
+ */
+export interface ResolveIntentResults {
+    targets: AppMetadata[];
+}
+
+/**
+ * the result of calling desktopAgent.resolveContext() is `Promise<ResolveContextResults>`
+ * the ResolveContextResults contains the list of intents for the given context type
+ */
+export interface ResolveContextResults {
+    intents: ResolveContextResult[];
+}
+
+/**
+ * a ResolveContextResult models an intent and supported app targets
+ */
+export interface ResolveContextResult {
+    intentName: IntentName;
+    targets: Array<AppMetadata>;
+}
+
+/**
+ * an FDC3 Intent
+ */
+export interface Intent {
```

Comment:
For me this doesn't address #12, as it means an intent is still and object that needs to be created.

It should just be "StartCall" + payload when you send it. I would like to separate the intent itself from the context (payload) _that is only provided as a parameter when sending it_

Reply:
The intent object always had a context property.  I removed context from the signature as I thought it was confusing and/or redundant.  We can add back the context arg but we should then remove the context property from the intent itself.

Reply:
The point is that the intent doesn't need a context property and therefore there is no need to have it as an object. What is the use case for having a context as a property on the intent when you create it?

The only time when you need to know the context is when sending the intent, making the context part of the intent reduces flexibility.

For example, app1 declares that it supports intent "StartChat" with context type "org.symphony.contact", whereas app2 declares that it supports "StartChat" with context type "com.db.contact". The commonality is the intent name, but the context type can be specific to each application's implementation.

So when I am declaring the common list of intents, I just call them "StartCall", "StartChat" etc. not caring about the context type.

But when I invoke the intent, it is in combination with a particular type of payload. I therefore agree that the context type shouldn't be both an argument in the call and a property on the intent, but I believe it to be the other way round: argument instead of property.

Happy to be persuaded otherwise/go with the majority on this, just trying to explain myself.